hypervisor/arch/x86/cpuid.c

```diff
@@ -158,9 +173,36 @@ static void init_vcpuid_entry(__unused struct vm *vm,
 		break;
 
 	default:
-		cpuid_subleaf(leaf, subleaf,
-				&entry->eax, &entry->ebx,
-				&entry->ecx, &entry->edx);
+		if (leaf >= 0x80000000U) {
+			/* unsupported extended cpuid level */
+			if (leaf > boot_cpu_data.extended_cpuid_level) {
+				entry->eax = 0U;
+				entry->ebx = 0U;
+				entry->ecx = 0U;
+				entry->edx = 0U;
+			} else {
+				cpuid_subleaf(leaf, subleaf,
+					&entry->eax, &entry->ebx,
+					&entry->ecx, &entry->edx);
+			}
+		} else if (leaf >= 0x40000000U) {
+			entry->eax = 0U;
```

Comment:
In this case, we don't present the leaf 0x400000000 ? or we only support 0x40000000U?

Reply:
remove it.